### PR TITLE
Config 

### LINKED ISSUE #730: Add config for marks properties that are not mapped to encoding channels
## Easy 

- [x] name, description should be added to the top level of the specification?  (name will be particular useful when we support interaction)
- [x] properties
  - [x] general:  `fillOpacity`/`strokeOpacity`, `strokeDash`, `strokeDashOffset`, `strokeWidth`, …
  - [x] line/area: `interpolate`, `tension`
  - [x] area/bar: `orient`
  - [x] text: `align`, `baseline`, `dx`, `dy`, `radius`, `theta`, `angle`, `font`, `fontSize`, `fontWeight`, `fontStyle` 

- [x] support specifying group mark’s properties (e.g., background, border, etc.)

## Hard 
- [ ] How should we deal with key, delay, ease?

### PR DESCRIPTION
**Pull request by kanitw**
#730 

- Add top-level `name`, `description`
- Add config: `background`, `scene` and `viewport` properties and reorder these top-level properties in the Vega output
- Split the config schema files, add interfaces
- Add config for mark properties

### CODE REVIEW

File: src/compiler/compiler.ts [hunk 1]
```diff
@@ -22,21 +22,24 @@ export function compile(spec, theme?) {
...
-               {field: layout.width.field} :
-               {value: layout.width},
-        height: layout.height.field ?
-                {field: layout.height.field} :
-                {value: layout.height}
+  let rootGroup:any = extend({
+      name: spec.name ? spec.name + '_root' : 'root',
+      type: 'group',
+    },
+    spec.description ? {description: spec.description} : {},
```
Comment (domoritz) [inline src/compiler/compiler.ts]: can you add more let? This is hard to read

File: src/compiler/compiler.ts [hunk 2]
```diff
@@ -70,14 +73,26 @@ export function compile(spec, theme?) {
...
-  var output = {
+  var output = extend(
+    spec.name ? {name: spec.name} : {},
+    {
       width: layout.width.field ? FIT : layout.width,
       height: layout.height.field ? FIT : layout.height,
-      padding: 'auto',
+      padding: 'auto'
+    },
+    ['viewport', 'background', 'scene'].reduce(function(topLevelConfig, property) {
```
Comment (domoritz) [inline src/compiler/compiler.ts]: can you put that in another variable?

File: src/compiler/compiler.ts (hunk 1 above)
Comment (kanitw) [inline src/compiler/compiler.ts]: I disagree.  The goal of this chunk of code is to optimize order of the vega output. 
While stylistically a little uglier, putting it like this allows you to determine order of the output in one glance.

File: src/compiler/compiler.ts (hunk 2 above)
Comment (kanitw) [inline src/compiler/compiler.ts]: I disagree.  The goal of this chunk of code is to optimize order of the vega output. 
While stylistically a little uglier, putting it like this allows you to determine order of the output in one glance.

**Pull request merged**